go-controller/pkg/libovsdbops/router.go

```diff
@@ -1005,7 +1005,10 @@ func GetRouterNATs(nbClient libovsdbclient.Client, router *nbdb.LogicalRouter) (
 	nats := []*nbdb.NAT{}
 	for _, uuid := range r.Nat {
 		nat, err := GetNAT(nbClient, &nbdb.NAT{UUID: uuid})
-		if err != nil && err != libovsdbclient.ErrNotFound {
+		if err == libovsdbclient.ErrNotFound {
```

Comment:
IIUC, we were hitting another error that wasn't "not found" error which we weren't handling?

also was this the only spot we need to fix? I am unsure if I reviewed the initial fix but we had other places where we did similar fixes right @flavio-fernandes ? -> let's make sure we double check those..
but based on the introduced commit hash: https://github.com/ovn-org/ovn-kubernetes/commit/25d892cc9e318d19738d9519e0e20fbab28f7eae this LGTM.

Reply:
I briefly scanned https://github.com/ovn-org/ovn-kubernetes/pull/3647/ last night and didn't see anything...but that was like after midnight soooo :)

Reply:
@tssurya @trozet I went over https://github.com/ovn-org/ovn-kubernetes/pull/3647 as well and dis not find any other places with the same misteka. ;)